lambda_art.py

```diff
@@ -0,0 +1,190 @@
+"""TODO: Put your header comment here."""
+
+import random
+import math
+from PIL import Image
+
+
+def build_random_function(min_depth, max_depth):
+    """Build a random function.
+
+    Builds a random function of depth at least min_depth and depth at most
+    max_depth. (See the assignment write-up for the definition of depth
+    in this context)
+
+    Args:
+        min_depth: the minimum depth of the random function
+        max_depth: the maximum depth of the random function
+
+    Returns:
+        The randomly generated function represented as a nested list.
+        (See the assignment writ-eup for details on the representation of
+        these functions)
+    """
+
+    if min_depth < 1:
+        var = random.randint(1,2)
+        if var == 1:
+            return lambda x,y: x
+        else:
+            return lambda x,y: y
+
+    min_depth -= 1
+    max_depth -= 1
+
+    #varies the depth of the function
+    d_range = max_depth - min_depth
+    if d_range > 0:
+        n = random.randint(1, d_range)
+        if n==1:
+            min_depth +=1
+
+    rand_int = random.randint(1,6)
+
+    if rand_int == 1:
+        inner = build_random_function(min_depth, max_depth)
+        return lambda x,y: math.cos(math.pi * inner(x,y))
+
+    if rand_int == 2:
+        inner = build_random_function(min_depth, max_depth)
+        return lambda x,y: math.sin(math.pi * inner(x,y))
+
+    if rand_int == 3:
+        inner = build_random_function(min_depth, max_depth)
+        return lambda x,y: inner(x,y) ** 3
+
+    if rand_int == 4:
+        inner1 = build_random_function(min_depth, max_depth)
+        inner2 = build_random_function(min_depth, max_depth)
+        return lambda x,y: inner1(x,y) * inner2(x,y)
+
+    if rand_int == 5:
+        inner1 = build_random_function(min_depth, max_depth)
+        inner2 = build_random_function(min_depth, max_depth)
+        return lambda x,y: .5*(inner1(x,y) + inner2(x,y))
+
+    if rand_int == 6:
+        inner = build_random_function(min_depth, max_depth)
+        return lambda x,y: math.sin(30 * math.pi * inner(x,y))
+
+    print('error')
```

Comment:
Check out https://docs.python.org/3/tutorial/errors.html#raising-exceptions - you can also raise errors, rather than printing and then returning nothing. Essentially, you can make a custom error if you want that degree of control over the program.

(Though, unless the built-in `random` library broke, this error should never appear - rand_int should always give you one of those values, so you should be fine either way.)